tools/system_libs.py

```diff
@@ -1724,6 +1724,50 @@ def variations(cls):
             [dict(malloc='emmalloc-verbose', **combo) for combo in combos if combo['verbose'] and not combo['memvalidate']])
 
 
+class libmimalloc(MTLibrary):
+  name = 'libmimalloc'
+
+  cflags = [
+    '-fno-builtin',
+    '-Wno-deprecated-pragma',
+    # build emmalloc as only a system allocator, without exporting itself onto
+    # malloc/free in the global scope
+    '-DEMMALLOC_NO_STD_EXPORTS',
+    # build mimalloc with an override of malloc/free
+    '-DMI_MALLOC_OVERRIDE',
+    # TODO: add build modes that include debug checks 1,2,3
+    '-DMI_DEBUG=0',
+  ]
+
+  # malloc/free/calloc are runtime functions and can be generated during LTO
+  # Therefor they cannot themselves be part of LTO.
+  force_object_files = True
+
+  includes = ['system/lib/mimalloc/include']
+
+  # Build all of mimalloc, and also emmalloc which is used as the system
+  # allocator underneath it.
+  src_dir = 'system/lib/'
```

Comment:
You can remove this line I think

Reply:
Unfortunately, not. Other libraries do this too. Without this it errors, as it uses this internally it seems. That might be worth working on, though, as it is surprising to me too.